<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>Step 1 - SAIGE</title> <meta name="Description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages."> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Step 1 | SAIGE</title> <meta name="generator" content="Jekyll v4.2.1" /> <meta property="og:title" content="Step 1" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages." /> <meta property="og:description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages." /> <link rel="canonical" href="http://localhost:4000/docs/set_step1.html" /> <meta property="og:url" content="http://localhost:4000/docs/set_step1.html" /> <meta property="og:site_name" content="SAIGE" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Step 1" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages.","headline":"Step 1","url":"http://localhost:4000/docs/set_step1.html"}</script> <!-- End Jekyll SEO tag --> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="http://localhost:4000/" class="site-title lh-tight"> SAIGE </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="http://localhost:4000/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/Installation.html" class="nav-list-link">Installation</a><ul class="nav-list "><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_bioconda.html" class="nav-list-link">Bioconda</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_conda.html" class="nav-list-link">Conda</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_docker.html" class="nav-list-link">Docker</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/single.html" class="nav-list-link">Single-variant test</a><ul class="nav-list "><li class="nav-list-item "><a href="http://localhost:4000/docs/single_step1.html" class="nav-list-link">Step 1</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/single_step2.html" class="nav-list-link">Step 2</a></li></ul></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/set.html" class="nav-list-link">Set-based test</a><ul class="nav-list "><li class="nav-list-item active"><a href="http://localhost:4000/docs/set_step1.html" class="nav-list-link active">Step 1</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/set_step2.html" class="nav-list-link">Step 2</a></li></ul></li><li class="nav-list-item"><a href="http://localhost:4000/docs/FAQ.html" class="nav-list-link">Frequently asked questions</a></li><li class="nav-list-item"><a href="http://localhost:4000/about/" class="nav-list-link">About</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search SAIGE" aria-label="Search SAIGE" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="http://localhost:4000/docs/set.html">Set-based test</a></li> <li class="breadcrumb-nav-list-item"><span>Step 1</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h1 id="step-1-fitting-the-null-logisticlinear-mixed-model"> <a href="#step-1-fitting-the-null-logisticlinear-mixed-model" class="anchor-heading" aria-labelledby="step-1-fitting-the-null-logisticlinear-mixed-model"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step 1: fitting the null logistic/linear mixed model </h1> <p>Step 1 for set-based tests are the same as for single-variant tests, except that</p> <ol> <li>Multiple variance ratios need to be estimated based on different minor allele count categories with <em>–isCateVarianceRatio=TRUE</em> <ul> <li>By default, two variance ratios are estiamted for 10 &lt;= MAC &lt; 20 and MAC &gt;= 20. Use <em>–cateVarRatioMinMACVecExclude</em> and <em>–cateVarRatioMaxMACVecInclude</em> to modify the MAC categories</li> <li>Note that the PLINK file need to contain at least 1000 variants whose MAC fall in these categories</li> </ul> </li> </ol> <p>2.1. When a full GRM is used to fit the null model (–useSparseGRMtoFitNULL=FALSE), variance ratios are estiamted based on both a full GRM (to be constrcuted using genotypes in the PLINK file) and a sparse GRM with <em>–useSparseGRMforVarRatio=TRUE</em></p> <ul> <li>Use –sparseGRMFile for the file containing the sparse GRM</li> <li>Use –sparseGRMSampleIDFile for the file containing the IDs for samples in the sparse GRM</li> </ul> <p>2.2. When a sparse GRM is used to fit the null model (–useSparseGRMtoFitNULL=TRUE), variance ratios are estiamted based on a sparse GRM and NULL (leave <em>–useSparseGRMforVarRatio=FALSE</em>)</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#Estimate categorical variance ratios (*--isCateVarianceRatio=TRUE*)
#Use a full GRM to fit the null model(by default, *--useSparseGRMtoFitNULL=FALSE*)
#Use an additional sparse GRM to estimate the variance ratio (*--useSparseGRMforVarRatio=TRUE*) 
Rscript step1_fitNULLGLMM.R     \
        --plinkFile=./input/nfam_100_nindep_0_step1_includeMoreRareVariants_poly_22chr  \
        --sparseGRMFile=output/sparseGRM_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseGRM.mtx   \
        --sparseGRMSampleIDFile=output/sparseGRM_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseGRM.mtx.sampleIDs.txt     \
        --useSparseGRMtoFitNULL=FALSE    \
        --phenoFile=./input/pheno_1000samples.txt_withdosages_withBothTraitTypes.txt \
        --phenoCol=y_binary \
        --covarColList=x1,x2,a9,a10 \
        --qCovarColList=a9,a10  \
        --sampleIDColinphenoFile=IID \
        --traitType=binary        \
        --outputPrefix=./output/example_binary \
        --nThreads=4    \
        --noEstFixedEff=TRUE    \
        --isCateVarianceRatio=TRUE      \
        --useSparseGRMforVarRatio=TRUE  \
        --IsOverwriteVarianceRatioFile=TRUE
</code></pre></div></div> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#Fit the null model using a sparse GRM (*--useSparseGRMtoFitNULL=TRUE*)
#Do not estiamte variance ratios (*--skipVarianceRatioEstimation=TRUE*), so in Step 2, the sparse GRM will be used directly in the tests
 
Rscript step1_fitNULLGLMM.R     \
        --sparseGRMFile=output/sparseGRM_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseGRM.mtx   \
        --sparseGRMSampleIDFile=output/sparseGRM_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseGRM.mtx.sampleIDs.txt     \
        --useSparseGRMtoFitNULL=TRUE    \
        --phenoFile=./input/pheno_1000samples.txt_withdosages_withBothTraitTypes.txt \
        --phenoCol=y_binary \
        --covarColList=x1,x2,a9,a10 \
        --qCovarColList=a9,a10  \
        --sampleIDColinphenoFile=IID \
        --traitType=binary        \
        --outputPrefix=./output/example_binary \
        --nThreads=2    \
        --noEstFixedEff=TRUE    \
        --skipVarianceRatioEstimation=TRUE
</code></pre></div></div> <h2 id="input-files"> <a href="#input-files" class="anchor-heading" aria-labelledby="input-files"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Input files </h2> <ol> <li> <p>(same as step 1 for single-variant assoc tests and step 0) <br /> Genotype file for constructing the genetic relationship matrix in the plink format <br /></p> </li> <li> <p>a file storing the sparse GRM (optional, output by step 0)</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> sparseGRM_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseGRM.mtx
</code></pre></div> </div> <p>The sparse GRM file can be opened using the <em>readMM</em> function in the R library <em>Matrix</em></p> </li> <li> <p>a file storing IDs of the samples in the sparse GRM (optional, output by step 0)</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> sparseGRM_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseGRM.mtx.sampleIDs.txt
</code></pre></div> </div> </li> </ol> <h2 id="output-files"> <a href="#output-files" class="anchor-heading" aria-labelledby="output-files"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Output files </h2> <ul> <li>same as the step 1 output by SAIGE for single-variant association tests <br /></li> </ul> <ol> <li>model file <br /></li> <li>association result file for the subset of randomly selected markers <br /></li> <li>variance ratio file <br /></li> </ol> <ul> <li>specific to SAIGE-GENE</li> </ul> <ol> <li> <p>sparse Sigma file</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> sparseGRM_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseSigma.mtx
</code></pre></div> </div> <p>NOTE: the file contains the sparse Sigma matrix, which is ** NOT ** the sparse GRM. The sparse Sigma matrix is computed based on the sparse GRM and the results of step 1.</p> </li> </ol> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
