<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>Step 2 - SAIGE</title> <meta name="Description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages."> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Step 2 | SAIGE</title> <meta name="generator" content="Jekyll v4.2.1" /> <meta property="og:title" content="Step 2" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages." /> <meta property="og:description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages." /> <link rel="canonical" href="http://localhost:4000/docs/set_step2.html" /> <meta property="og:url" content="http://localhost:4000/docs/set_step2.html" /> <meta property="og:site_name" content="SAIGE" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Step 2" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages.","headline":"Step 2","url":"http://localhost:4000/docs/set_step2.html"}</script> <!-- End Jekyll SEO tag --> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="http://localhost:4000/" class="site-title lh-tight"> SAIGE </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="http://localhost:4000/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/Installation.html" class="nav-list-link">Installation</a><ul class="nav-list "><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_bioconda.html" class="nav-list-link">Bioconda</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_conda.html" class="nav-list-link">Conda</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_docker.html" class="nav-list-link">Docker</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/single.html" class="nav-list-link">Single-variant test</a><ul class="nav-list "><li class="nav-list-item "><a href="http://localhost:4000/docs/single_step1.html" class="nav-list-link">Step 1</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/single_step2.html" class="nav-list-link">Step 2</a></li></ul></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/set.html" class="nav-list-link">Set-based test</a><ul class="nav-list "><li class="nav-list-item "><a href="http://localhost:4000/docs/set_step1.html" class="nav-list-link">Step 1</a></li><li class="nav-list-item active"><a href="http://localhost:4000/docs/set_step2.html" class="nav-list-link active">Step 2</a></li></ul></li><li class="nav-list-item"><a href="http://localhost:4000/docs/FAQ.html" class="nav-list-link">Frequently asked questions</a></li><li class="nav-list-item"><a href="http://localhost:4000/about/" class="nav-list-link">About</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search SAIGE" aria-label="Search SAIGE" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="http://localhost:4000/docs/set.html">Set-based test</a></li> <li class="breadcrumb-nav-list-item"><span>Step 2</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h2 id="step-2-performing-the-region--or-gene-based-association-tests"> <a href="#step-2-performing-the-region--or-gene-based-association-tests" class="anchor-heading" aria-labelledby="step-2-performing-the-region--or-gene-based-association-tests"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step 2: performing the region- or gene-based association tests </h2> <ul> <li>The command line is the same as the step 2 for single-variant assoc tests, except that a group file is specified (–groupFile)</li> <li>Each line is for one gene/set of variants. The first element is for gene/set name.</li> <li>The rest of the line is for variant ids included in this gene/set. For vcf/sav, the genetic marker ids are in the format chr:pos_ref/alt. For bgen, the genetic marker ids should match the ids in the bgen file. Each element in the line is separated by tab. <strong>The marker ids in the group file for vcf/sav need to be sorted by chr and pos.</strong></li> <li>IsSingleVarinGroupTest=TRUE is to perform single-variant association tests as well for markers included in the gene-based tests</li> <li>–IsOutputBETASEinBurdenTest is to output effect sizes for burden tests (this option is still under development)</li> <li> <p>Same as the single-variant association tests, conditional analysis based summary stats can be performed (–condition) can be performed in step 2 with dosage/genotype input file formats VCF, BGEN and SAV.</p> </li> <li>Use <em>–function_group_test</em> to list different annotations</li> <li>Use <em>–maxMAFforGroupTest</em> for different max MAF cutoffs</li> <li> <p>By default, SKAT-O, SKAT, and BURDEN tests are performed</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
      --bgenFile=./input/genotype_100markers.bgen    \
      --bgenFileIndex=./input/genotype_100markers.bgen.bgi \
      --SAIGEOutputFile=./genotype_100markers_bgen_groupTest_out.txt \
      --chrom=1 \
      --LOCO=TRUE    \
      --AlleleOrder=ref-first \
      --minMAF=0 \
      --minMAC=0.5 \
      --sampleFile=./input/samplelist.txt \
      --GMMATmodelFile=./output/example_binary_fullGRM.rda \
      --varianceRatioFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt   \
      --numLinesOutput=10     \
      --groupFile=./input/group_new_snpid.txt \
      --sparseSigmaFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseSigma.mtx       \
      --function_group_test="lof,missense;lof,missense;lof;synonymous"        \
      --maxMAFforGroupTest=0.0001,0.001,0.01
</code></pre></div> </div> </li> <li> <p>Specify <em>–r.corr=1</em> to only perform BURDEN test</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
      --bgenFile=./input/genotype_100markers.bgen    \
      --bgenFileIndex=./input/genotype_100markers.bgen.bgi \
      --SAIGEOutputFile=./output/genotype_100markers_bgen_groupTest_onlyBURDEN.out.txt \
      --chrom=1 \
      --LOCO=TRUE    \
      --AlleleOrder=ref-first \
      --minMAF=0 \
      --minMAC=0.5 \
      --sampleFile=./input/samplelist.txt \
      --GMMATmodelFile=./output/example_binary_fullGRM.rda \
      --varianceRatioFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt   \
      --numLinesOutput=10     \
      --groupFile=./input/group_new_snpid.txt \
      --sparseSigmaFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseSigma.mtx       \
      --function_group_test="lof,missense;lof,missense;lof;synonymous"        \
      --maxMAFforGroupTest=0.0001,0.001,0.01  \
      --r.corr=1
</code></pre></div> </div> </li> <li> <p>Use –condition= to perform conditioning analysis</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
      --bgenFile=./input/genotype_100markers.bgen    \
      --bgenFileIndex=./input/genotype_100markers.bgen.bgi \
      --SAIGEOutputFile=./genotype_100markers_bgen_groupTest_conditional.out.txt \
      --chrom=1 \
      --LOCO=TRUE    \
      --AlleleOrder=ref-first \
      --minMAF=0 \
      --minMAC=0.5 \
      --sampleFile=./input/samplelist.txt \
      --GMMATmodelFile=./output/example_binary_fullGRM.rda \
      --varianceRatioFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt   \
      --numLinesOutput=10     \
      --groupFile=./input/group_new_snpid.txt \
      --sparseSigmaFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseSigma.mtx       \
      --function_group_test="lof,missense;lof,missense;lof;synonymous"        \
      --maxMAFforGroupTest=0.0001,0.001,0.01  \
      --condition=rs30,rs79
</code></pre></div> </div> </li> <li> <p>PLINK file as input</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
      --bedFile=./input/genotype_100markers.bed       \
      --bimFile=./input/genotype_100markers.bim       \
      --famFile=./input/genotype_100markers.fam       \
      --SAIGEOutputFile=./output/genotype_100markers_plink_groupTest_out.txt \
      --chrom=1 \
      --LOCO=TRUE    \
      --AlleleOrder=alt-first \
      --minMAF=0 \
      --minMAC=0.5 \
      --sampleFile=./input/samplelist.txt \
      --GMMATmodelFile=./output/example_binary_fullGRM.rda \
      --varianceRatioFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt   \
      --numLinesOutput=10     \
      --groupFile=./input/group_new_snpid.txt \
      --sparseSigmaFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseSigma.mtx       \
      --function_group_test="lof,missense;lof,missense;lof;synonymous"        \
      --maxMAFforGroupTest=0.0001,0.001,0.01
</code></pre></div> </div> </li> <li> <p>VCF file as input</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
      --vcfFile=./input/genotype_100markers.vcf.gz    \
      --vcfFileIndex=./input/genotype_100markers.vcf.gz.csi     \
      --vcfField=GT   \
      --SAIGEOutputFile=./output/genotype_100markers_vcf_groupTest_out.txt \
      --chrom=1 \
      --LOCO=TRUE    \
      --AlleleOrder=alt-first \
      --minMAF=0 \
      --minMAC=0.5 \
      --sampleFile=./input/samplelist.txt \
      --GMMATmodelFile=./output/example_binary_fullGRM.rda \
      --varianceRatioFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt   \
      --numLinesOutput=10     \
      --groupFile=./input/group_new_chrposa1a2.txt \
      --sparseSigmaFile=./output/example_binary_fullGRM_sparseGRM_categorical_varRatio.varianceRatio.txt_relatednessCutoff_0.125_1000_randomMarkersUsed.sparseSigma.mtx       \
      --function_group_test="lof,missense;lof,missense;lof;synonymous"        \
      --maxMAFforGroupTest=0.0001,0.001,0.01
</code></pre></div> </div> </li> </ul> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
