<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>Step 2 - SAIGE</title> <meta name="Description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages."> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Step 2 | SAIGE</title> <meta name="generator" content="Jekyll v4.2.1" /> <meta property="og:title" content="Step 2" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages." /> <meta property="og:description" content="Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages." /> <link rel="canonical" href="http://localhost:4000/docs/single_step2.html" /> <meta property="og:url" content="http://localhost:4000/docs/single_step2.html" /> <meta property="og:site_name" content="SAIGE" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Step 2" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages.","headline":"Step 2","url":"http://localhost:4000/docs/single_step2.html"}</script> <!-- End Jekyll SEO tag --> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="http://localhost:4000/" class="site-title lh-tight"> SAIGE </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="http://localhost:4000/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/Installation.html" class="nav-list-link">Installation</a><ul class="nav-list "><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_bioconda.html" class="nav-list-link">Bioconda</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_conda.html" class="nav-list-link">Conda</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/Installation_docker.html" class="nav-list-link">Docker</a></li></ul></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/single.html" class="nav-list-link">Single-variant test</a><ul class="nav-list "><li class="nav-list-item "><a href="http://localhost:4000/docs/single_step1.html" class="nav-list-link">Step 1</a></li><li class="nav-list-item active"><a href="http://localhost:4000/docs/single_step2.html" class="nav-list-link active">Step 2</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="http://localhost:4000/docs/set.html" class="nav-list-link">Set-based test</a><ul class="nav-list "><li class="nav-list-item "><a href="http://localhost:4000/docs/set_step1.html" class="nav-list-link">Step 1</a></li><li class="nav-list-item "><a href="http://localhost:4000/docs/set_step2.html" class="nav-list-link">Step 2</a></li></ul></li><li class="nav-list-item"><a href="http://localhost:4000/docs/FAQ.html" class="nav-list-link">Frequently asked questions</a></li><li class="nav-list-item"><a href="http://localhost:4000/about/" class="nav-list-link">About</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search SAIGE" aria-label="Search SAIGE" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="http://localhost:4000/docs/single.html">Single-variant test</a></li> <li class="breadcrumb-nav-list-item"><span>Step 2</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h1 id="step-2-performing-single-variant-association-tests"> <a href="#step-2-performing-single-variant-association-tests" class="anchor-heading" aria-labelledby="step-2-performing-single-variant-association-tests"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step 2: performing single-variant association tests </h1> <ul> <li>For <strong>binary traits</strong>, saddle point approximation is used to account for case-control imbalance.</li> <li>File formats for dosages/genotypes of genetic variants to be tested can be used: PLINK, VCF, <a href="https://bitbucket.org/gavinband/bgen/overview">BGEN</a>, <a href="https://github.com/statgen/savvy">SAV</a></li> <li>Conditional analysis based summary stats can be performed (–condition) can be performed in Step 2</li> <li>To query and test a subset of markers</li> <li> <ul> <li>both variant IDs and range of chromosome positions can be specified for BGEN input (–idstoIncludeFile, –rangestoIncludeFile)</li> </ul> </li> <li>If LOCO=TRUE, –chrom MUST be specified</li> <li>For VCF/BCF/SAV input, –vcfField=DS to test dosages and –vcfField=GT to test genotypes</li> <li>Samples with missing genotypes/dosages can be imputed with the mean dosages (–impute_method=mean) or as minor allele homozygotes (–impute_method=minor). Note that currently dropping samples with missing genotypes/dosages is not supported</li> <li>–sampleFile is used specify a file with sample IDs for bgen file. Please <strong>DO NOT</strong> include a header in the file.</li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#check the help info for step 2
Rscript step2_SPAtests.R --help
</code></pre></div></div> <h2 id="single-variant-association-tests"> <a href="#single-variant-association-tests" class="anchor-heading" aria-labelledby="single-variant-association-tests"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Single-variant association tests </h2> <ul> <li> <p>For binary traits, use <em>–is_output_moreDetails=TRUE</em> to output heterozygous and homozygous counts in cases and controls</p> </li> <li> <p>Using <em>–bgenFile, –bgenFileIndex, –AlleleOrder, –sampleFile</em> for bgen input</p> </li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
        --bgenFile=./input/genotype_100markers.bgen    \
        --bgenFileIndex=./input/genotype_100markers.bgen.bgi \
        --sampleFile=./input/samplelist.txt \
        --AlleleOrder=ref-first \
        --SAIGEOutputFile=./genotype_100markers_single_out.txt \
        --chrom=1 \
        --LOCO=TRUE    \
        --minMAF=0 \
        --minMAC=20 \
        --GMMATmodelFile=./output/example_binary.rda \
        --varianceRatioFile=./output/example_binary.varianceRatio.txt   \
        --numLinesOutput=10	\
	
</code></pre></div></div> <ul> <li>Using <em>–bedFile, –bimFile, –famFile, –AlleleOrder</em> for PLINK input</li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
        --bedFile=./input/genotype_100markers.bed       \
        --bimFile=./input/genotype_100markers.bim       \
        --famFile=./input/genotype_100markers.fam       \
        --AlleleOrder=alt-first \
        --SAIGEOutputFile=./output/new_single_plink.txt \
        --chrom=1       \
        --LOCO=TRUE     \
        --minMAF=0 \
        --minMAC=20 \
        --GMMATmodelFile=./output/example_binary.rda \
        --varianceRatioFile=./output/example_binary.varianceRatio.txt   \
        --is_output_moreDetails=TRUE    \
</code></pre></div></div> <ul> <li>Using <em>–vcfFile， –vcfFileIndex， –vcfField</em> for VCF, BCF, and SAV input</li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
        --vcfFile=./input/genotype_100markers.vcf.gz    \
        --vcfFileIndex=./input/genotype_100markers.vcf.gz.csi     \
        --vcfField=GT   \
        --SAIGEOutputFile=./output/new_single_vcf.txt \
        --chrom=1       \
        --LOCO=TRUE     \
        --minMAF=0 \
        --minMAC=20 \
        --GMMATmodelFile=./output/example_binary.rda \
        --varianceRatioFile=./output/example_binary.varianceRatio.txt   \
        --is_output_moreDetails=TRUE
</code></pre></div></div> <h2 id="conditional-analysis"> <a href="#conditional-analysis" class="anchor-heading" aria-labelledby="conditional-analysis"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Conditional analysis </h2> <ul> <li>–condition = Genetic marker ids (<strong>chr:pos_ref/alt for VCF,BCF,SAV or marker id for PLINK and BGEN</strong>) separated by comma. e.g.chr3:101651171_C/T,chr3:101651186_G/A</li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
        --vcfFile=./input/genotype_100markers.vcf.gz    \
        --vcfFileIndex=./input/genotype_100markers.vcf.gz.csi     \
        --vcfField=GT   \
        --SAIGEOutputFile=./output/new_single_vcf.txt \
        --chrom=1       \
        --LOCO=TRUE     \
        --minMAF=0 \
        --minMAC=20 \
        --GMMATmodelFile=./output/example_binary.rda \
        --varianceRatioFile=./output/example_binary.varianceRatio.txt   \
        --is_output_moreDetails=TRUE    \
        --condition=1:13_A/C,1:79_A/C
</code></pre></div></div> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step2_SPAtests.R        \
        --bgenFile=./input/genotype_100markers.bgen    \
        --bgenFileIndex=./input/genotype_100markers.bgen.bgi \
        --SAIGEOutputFile=./genotype_100markers_single_out.txt \
        --chrom=1 \
        --LOCO=TRUE    \
        --AlleleOrder=ref-first \
        --minMAF=0 \
        --minMAC=1 \
        --sampleFile=./input/samplelist.txt \
        --GMMATmodelFile=./output/example_binary.rda \
        --varianceRatioFile=./output/example_binary.varianceRatio.txt   \
        --numLinesOutput=10     \
        --condition=rs13,rs79
</code></pre></div></div> <h2 id="input-files"> <a href="#input-files" class="anchor-heading" aria-labelledby="input-files"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Input files </h2> <ol> <li>Dosage file SAIGE supports different formats for dosages: PLINK, VCF, BCF, <a href="http://www.well.ox.ac.uk/~gav/bgen_format/bgen_format_v1.2.html">BGEN</a> and <a href="https://github.com/statgen/savvy">SAV</a>. <ul> <li>PLINK</li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> ./input/genotype_100markers.bed
 ./input/genotype_100markers.bim
 ./input/genotype_100markers.fam
</code></pre></div> </div> </li> </ol> <ul> <li> <p>BGEN</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  genotype_100markers.bgen
  genotype_100markers.bgen.bgi 
</code></pre></div> </div> </li> <li> <p>VCF containing genotypes</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  zcat genotype_10markers.vcf.gz | less -S
  genotype_10markers.vcf.gz.tbi
</code></pre></div> </div> </li> <li> <p>VCF containing dosages</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  zcat dosage_10markers.vcf.gz | less -S
  dosage_10markers.vcf.gz.tbi
</code></pre></div> </div> </li> <li> <p>SAV</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  dosage_10markers.sav
  dosage_10markers.sav.s1r
</code></pre></div> </div> </li> </ul> <ol> <li> <p>Sample file &lt;br&gt; This file contains one column for sample IDs corresponding to the sample order in the dosage file. <strong>No header is included.</strong> The option was originally for BGEN file that does not contain sample information.</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> less -S sampleIDindosage.txt
</code></pre></div> </div> </li> <li> <p>Model file from step 1</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> ./output/example.rda
</code></pre></div> </div> </li> <li> <p>Variance ratio file from step 1</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> ./output/example.varianceRatio.txt
</code></pre></div> </div> </li> </ol> <h2 id="output-file"> <a href="#output-file" class="anchor-heading" aria-labelledby="output-file"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Output file </h2> <p>A file with association test results</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>```
less -S ./output/example.SAIGE.vcf.dosage.txt
``` NOTE: * Association results are with regard to Allele2.  * For binary traits, the header of the output file:  ``` CHR POS SNPID Allele1 Allele2 AC_Allele2 AF_Allele2 imputationInfo N BETA SE Tstat p.value p.value.NA Is.SPA.converge varT varTstar AF.Cases AF.Controls ``` * For quantitative traits, the header of the output file:  ``` CHR POS SNPID Allele1 Allele2 AC_Allele2 AF_Allele2 imputationInfo N BETA SE Tstat p.value varT varTstar```
</code></pre></div></div> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>CHR: chromosome
POS: genome position 
SNPID: variant ID
Allele1: allele 1
Allele2: allele 2
AC_Allele2: allele count of allele 2
AF_Allele2: allele frequency of allele 2
imputationInfo: imputation info. If not in dosage/genotype input file, will output 1
N: sample size
BETA: effect size of allele 2
SE: standard error of BETA
Tstat: score statistic of allele 2
p.value: p value (with SPA applied for binary traits)
p.value.NA: p value when SPA is not applied (only for binary traits)
Is.SPA.converge: whether SPA is converged or not (only for binary traits)
varT: estimated variance of score statistic with sample relatedness incorporated
varTstar: variance of score statistic without sample relatedness incorporated
AF.Cases: allele frequency of allele 2 in cases (only for binary traits and if --IsOutputAFinCaseCtrl=TRUE)
AF.Controls: allele frequency of allele 2 in controls (only for binary traits and if --IsOutputAFinCaseCtrl=TRUE)
Tstat_cond, p.value_cond, varT_cond, BETA_cond, SE_cond: summary stats for conditional analysis
</code></pre></div></div> <h4 id="an-example-with-a-signal"> <a href="#an-example-with-a-signal" class="anchor-heading" aria-labelledby="an-example-with-a-signal"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> An example with a signal </h4> <ul> <li>p.value in Step 2 for the marker is ~4.16 x 10^-7</li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Rscript step1_fitNULLGLMM.R     \
        --plinkFile=./input/nfam_100_nindep_0_step1_includeMoreRareVariants_poly \
        --phenoFile=./input/Prev_0.1_nfam_1000.pheno_positive_pheno.txt \
        --phenoCol=y \
        --covarColList=x1,x2 \
        --sampleIDColinphenoFile=IID \
        --traitType=binary        \
        --outputPrefix=./output/example_binary_positive_signal \
        --nThreads=4    \
        --LOCO=FALSE    \
        --minMAFforGRM=0.01

Rscript step2_SPAtests.R        \
        --vcfFile=./input/nfam_1000_MAF0.2_nMarker1_nseed200.vcf.gz \
        --vcfFileIndex=./input/nfam_1000_MAF0.2_nMarker1_nseed200.vcf.gz.tbi \
        --vcfField=GT \
        --chrom=1 \
        --minMAF=0.0001 \
        --minMAC=1 \
        --GMMATmodelFile=./output/example_binary_positive_signal.rda \
        --varianceRatioFile=./output/example_binary_positive_signal.varianceRatio.txt \
        --SAIGEOutputFile=./output/example_binary_positive_signal.assoc.step2.txt \
        --numLinesOutput=2 \
        --IsOutputAFinCaseCtrl=TRUE     \
</code></pre></div></div> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
